### PR TITLE
Add tests to ensure that admission matches the CLI

### PR DESCRIPTION
**Pull request by deads2k**
Add testing for the admission server.  This re-uses the same test definition and comparison handling as the CLI tests to ensure equivalence of the two.

/assign @JoelSpeed

### CODE REVIEW

File: pkg/cmd/checkadmission/admission_tester.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	if body, err := result.Raw(); err != nil {
+		t.Log(string(body))
+		t.Fatal(err)
+	}
+	err = result.Into(actual)
+	if err != nil {
+		t.Fatal(err)
+	}
+	actual.Request = nil
+	//t.Log(spew.Sdump(actual))
```
Comment (JoelSpeed) [inline pkg/cmd/checkadmission/admission_tester.go:89]: Debug, remove before merge?

File: pkg/cmd/checkadmission/admission_tester.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			}
+			results := comparatorNameToResults[name]
+			results.Name = name
+			results.Errors = append(results.Errors, cause.Message)
+			comparatorNameToResults[name] = results
+		}
+	}
+
+	for _, warning := range actual.Response.Warnings {
+		if strings.HasPrefix(warning, "fyi: ") {
```
Comment (JoelSpeed) [inline pkg/cmd/checkadmission/admission_tester.go:110]: fyi? Are the errors coming out as an FYI?

File: pkg/cmd/checkadmission/admission_tester.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	for name := range comparatorNameToResults {
+		results := comparatorNameToResults[name]
+		actualResults = append(actualResults, results)
+	}
+
+	if len(actualResults) > 1 {
+		panic(len(actualResults))
+	}
```
Comment (JoelSpeed) [inline pkg/cmd/checkadmission/admission_tester.go:134]: Why can there not be more than one result?

File: pkg/cmd/checkadmission/check_admission_test.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+	"k8s.io/apiserver/pkg/features"
+	"k8s.io/apiserver/pkg/util/feature"
+
+	"github.com/openshift/crd-schema-checker/pkg/manifestcomparators"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	//clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
```
Comment (JoelSpeed) [inline pkg/cmd/checkadmission/check_admission_test.go:20]: Remove before merge?

File: pkg/cmd/checkadmission/check_admission_test.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api/v1"
+	"k8s.io/klog/v2"
+)
+
+func TestAdmissionServer(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (JoelSpeed) [reply pkg/cmd/checkadmission/check_admission_test.go:25]: I guess there's no easy library that you want to import that can do this for us?

File: pkg/cmd/checkadmission/admission_tester.go (hunk 3 above)
Comment (deads2k) [reply pkg/cmd/checkadmission/admission_tester.go:134]: removing.

**Comment by deads2k** [comment]
> I guess there's no easy library that you want to import that can do this for us?

No.  Or rather not at the moment.  In theory this could (should?) be provided from the genericadmissionserver .

**Comment by deads2k** [comment]
Updated for comments.

**Review by JoelSpeed — COMMENTED** [review]
/lgtm

**Pull request merged**